### PR TITLE
fix: Refresh Sell screen when updating the current draft submission

### PR DESCRIPTION
**Pull request by olerichter00**
### Description

With this fix, the Sell screen refetches the current draft submission when clicking "Save & Exit" in the submission form.

### Before

https://github.com/artsy/eigen/assets/4691889/437ee2c7-7fb2-4dcf-aac5-68beb65e74ce

### After

https://github.com/artsy/eigen/assets/4691889/763da21c-e30f-4ff0-9ce5-f658027c933e

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Refresh Sell screen when updating the current draft submission (behind ff) - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 68e1870e96b0a535f33633d9d248a8c3911e880e
</p>

File: src/app/Scenes/SellWithArtsy/ArtworkForm/Components/SubmitArtworkTopNavigation.tsx [hunk 1]
```diff
@@ -101,9 +104,7 @@ export const SubmitArtworkTopNavigation: React.FC<{}> = () => {
               }, 100)
             }}
           />
-        )}
-
-        {currentStep !== "SelectArtist" && (
+        ) : (
```
Comment (olerichter00) [inline src/app/Scenes/SellWithArtsy/ArtworkForm/Components/SubmitArtworkTopNavigation.tsx:107]: Just refactoring this :)

File: src/app/Scenes/SellWithArtsy/SellWithArtsyHome.tsx [hunk 2]
```diff
@@ -22,31 +19,46 @@ import { Testimonials } from "app/Scenes/SellWithArtsy/Components/Testimonials"
...
+  useEffect(() => {
+    RefreshEvents.addListener(SELL_SCREEN_REFRESH_KEY, handleRefreshEvent)
+    return () => {
+      RefreshEvents.removeListener(SELL_SCREEN_REFRESH_KEY, handleRefreshEvent)
+    }
+  }, [])
+
+  const handleRefreshEvent = () => {
+    increaseFetchKey()
+  }
```
Comment (olerichter00) [inline src/app/Scenes/SellWithArtsy/SellWithArtsyHome.tsx:51]: Not sure if there are better solutions to refresh the screen 🤔

But we use these listeners in a couple of places, and they work.

**Review by MounirDhahri — APPROVED** [review]
all good - i think this is fine already.

**Pull request merged**